narwhal/consensus/src/bullshark.rs

```diff
@@ -374,4 +309,35 @@ impl Bullshark {
 
         Ok((Outcome::Commit, committed_sub_dags))
     }
+
+    fn report_leader_on_time_metrics(&mut self, certificate_round: Round, state: &ConsensusState) {
+        // Under normal circumstances every odd round should trigger leader election for its previous
+        // even round. We consider a "hit" in this case when the leader has been elected when the network
+        // has not moved to the next even round (so latency it's still in the expected range). If the network
+        // has moved to the next even round and the leader has not been elected/committed, then we consider
+        // this a "miss". The leader might be committed later on, but we don't consider this a case where
+        // the leader has been committed "on time".
+        if certificate_round > self.max_inserted_certificate_round
+            && certificate_round % 2 == 0
+            && certificate_round > 2
+        {
+            let previous_leader_round = certificate_round - 2;
+            let authority = Self::leader_authority(&self.committee, previous_leader_round);
+
+            if state.last_round.committed_round < previous_leader_round {
+                self.metrics
+                    .leader_commit_on_time
```

Comment:
what do you think about calling the label `leader_commit_accuracy` to go along with the label values of "hit"/"miss". OR `leader_commit_timeliness`?